File: manual/code/test_suites/python/python/test_funcy_distinct.py
import unittest
from function_funcy_distinct import ListProcessor

class TestFuncyDistinct(unittest.TestCase):

    def setUp(self):
        self.processor = ListProcessor()

    def test_remove_duplicates(self):
        result = self.processor.process_list("1,2,3,1,4,2,5")
        self.assertEqual(result, "1,2,3,4,5")

    def test_all_unique_elements(self):
        result = self.processor.process_list("1,2,3,4,5")
        self.assertEqual(result, "1,2,3,4,5")

    def test_mixed_types(self):
        result = self.processor.process_list("1,apple,1,banana,apple,3.14,3.14")
        self.assertEqual(result, "1,apple,banana,3.14")

    def test_mixed_case_strings(self):
        result = self.processor.process_list("apple,Apple,banana,apple,Banana")
        self.assertEqual(result, "apple,Apple,banana,Banana")

    def test_large_list_with_duplicates(self):
        input_str = ",".join(["1"] * 1000 + ["2"] * 1000 + ["3"] * 1000)
        result = self.processor.process_list(input_str)
        self.assertEqual(result, "1,2,3")

if __name__ == '__main__':
    unittest.main()

File: data/java/tests/test_funcy_distinct.java
package com.example;

import static org.junit.jupiter.api.Assertions.assertEquals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

public class test_funcy_distinct {

    private function_funcy_distinct processor;

    @BeforeEach
    public void setUp() {
        processor = new function_funcy_distinct();
    }

    @Test
    public void test_remove_duplicates() {
        String result = processor.process_list("1,2,3,1,4,2,5");
        assertEquals("1,2,3,4,5", result);
    }

    @Test
    public void test_all_unique_elements() {
        String result = processor.process_list("1,2,3,4,5");
        assertEquals("1,2,3,4,5", result);
    }

    @Test
    public void test_mixed_types() {
        String result = processor.process_list("1,apple,1,banana,apple,3.14,3.14");
        assertEquals("1,apple,banana,3.14", result);
    }

    @Test
    public void test_mixed_case_strings() {
        String result = processor.process_list("apple,Apple,banana,apple,Banana");
        assertEquals("apple,Apple,banana,Banana", result);
    }

    @Test
    public void test_large_list_with_duplicates() {
        String input_str = String.join(",", java.util.Collections.nCopies(1000, "1")) + "," +
                String.join(",", java.util.Collections.nCopies(1000, "2")) + "," +
                String.join(",", java.util.Collections.nCopies(1000, "3"));
        String result = processor.process_list(input_str);
        assertEquals("1,2,3", result);
    }
}